### PR TITLE
Fix handling of deeply nested charts_dir

### LINKED ISSUE #46: Always fail to find changed charts with deeply nested (>2 levels) `charts_dir`
It looks as though this code will fail when the charts directory is nested below another directory: https://github.com/helm/chart-releaser-action/blob/06d81fa7f63ab865cf0ed2045bde38529c392845/cr.sh#L202-L209

For an example of this, see rook/rook, where `charts_dir` is `cluster/charts`:

* [Structure](https://github.com/LukeCarrier/rook/tree/nfs-chart/cluster/charts)
* [Failed workflow](https://github.com/LukeCarrier/rook/runs/1265065823)

Would you accept a PR to fix this?

### PR DESCRIPTION
**Pull request by LukeCarrier**
Fixes #46

### CODE REVIEW

File: cr.sh [hunk 1]
```diff
@@ -199,12 +199,8 @@ lookup_changed_charts() {
     local changed_files
     changed_files=$(git diff --find-renames --name-only "$commit" -- "$charts_dir")
 
-    local fields
-    if [[ "$charts_dir" == '.' ]]; then
-        fields='1'
-    else
-        fields='1,2'
-    fi
+    local depth=$(( $(tr "/" "\n" <<< "$charts_dir" | wc -l) + 1))
```
Comment (unguiculus) [inline cr.sh:202]:
```suggestion
    local depth=$(( $(tr "/" "\n" <<< "$charts_dir" | wc -l) + 1 ))
```

**Review by unguiculus — CHANGES_REQUESTED** [review]
That's a pretty clever solution. Thanks.

**Comment by unguiculus** [comment]
cc @scottrigby @davidkarlsen

File: cr.sh (hunk 1 above)
Comment (jsoref) [reply cr.sh:202]: Not sure if people have specific preferences... one could save the extra math operation by trading for an extra echo:
```suggestion
    local depth=$( (echo; tr "/" "\n" <<< "$charts_dir") | wc -l)
```

Comment (LukeCarrier) [reply cr.sh:202]: I'm happy with either change; I thought direct math more clearly expressed intent but I appreciate that it's a little used bit of Bash syntax.

Comment (unguiculus) [reply cr.sh]: I agree, direct math is better.

**Review by unguiculus — APPROVED** [review]

**Review by scottrigby — APPROVED** [review]

**Comment by itewk** [comment]
This fix seems to have broken things. when using the @master of this action now only the first chart in my REPO/charts/* folder is getting released....

**Comment by itewk** [comment]
> This fix seems to have broken things. when using the @master of this action now only the first chart in my REPO/charts/* folder is getting released....

the issue turns out to be absolute vs relative paths. i was passing in an absolute path to my charts dir, that worked with `v1.0.0` but with `v1.1.1` with this change the `charts_dir` has to be a relative path

**Comment by monotek** [comment]
Using "." as chart dir is broken too :(
https://github.com/helm/chart-releaser-action/issues/61

**Pull request merged**